### PR TITLE
Set permissions on files/directories during creation (+ small warning/analyzer fixes)

### PR DESCRIPTION
**Pull request by CarterA**
Pretty straightforward. This just lets you set permissions on files and directories during creation. I made it use an NSNumber storing an unsigned long because that's what I saw ConnectionKit doing. Let me know if there's anything you want me to do differently.

### CODE REVIEW

File: CURLHandleSource/CURLFTPSession.m [hunk 1]
```diff
@@ -193,7 +195,10 @@ - (BOOL)createDirectoryAtPath:(NSString *)path withIntermediateDirectories:(BOOL
     
     // Custom command to delete the file once we're in the correct directory
     // CURLOPT_PREQUOTE does much the same thing, but sometimes runs the delete command twice in my testing
-    [request curl_setPostTransferCommands:[NSArray arrayWithObject:[@"MKD " stringByAppendingString:[path lastPathComponent]]]];
+	NSMutableArray *postTransferCommands = [NSMutableArray arrayWithObject:[@"MKD " stringByAppendingString:[path lastPathComponent]]];
+	if (permissions)
+		[postTransferCommands addObject:[NSString stringWithFormat:@"SITE CHMOD %lu %@", [permissions unsignedLongValue], [path lastPathComponent]]];
+    [request curl_setPostTransferCommands:postTransferCommands];
```
Comment (mikeabdullah) [inline CURLHandleSource/CURLFTPSession.m:201]: One downside here is that if intermediate directories need to be created, they won't have permissions applied to them. Is that OK with you?

Comment (CarterA) [inline CURLHandleSource/CURLFTPSession.m:201]: Ah, I never ran into that with my use of the class, but I haven't been creating intermediates. Adding that would probably be a good idea, though it'll be rather inefficient.

Comment (mikeabdullah) [inline CURLHandleSource/CURLFTPSession.m:201]: I don't think it's especially inefficient. But does mean we'd have to re-implement the logic that libcurl already provides of creating intermediate directories.

**Comment by CarterA** [comment]
Splitting this off into a separate method makes _way_ more sense, you're right. I'm going to close this now.

**Pull request closed without merging**